Flake on jit_test.go: TestAmd64Compiler_compileBrTable

at the time, the code that failed was this line, which was nil.

```
	jitcall(
		uintptr(unsafe.Pointer(&code[0])),
		uintptr(unsafe.Pointer(j.vm)),
	)
```

It should not be nil due to jit_amd64.s
```
// jitcall(codeSegment, callEngine)
TEXT ·jitcall(SB),NOSPLIT|NOFRAME,$0-16
--snip--
```

```
--- FAIL: TestAmd64Compiler_compileBrTable (0.39s)
[18](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:18)
    --- FAIL: TestAmd64Compiler_compileBrTable/index_register=2074,tmpRegister=2064 (0.02s)
[19](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:19)
        --- FAIL: TestAmd64Compiler_compileBrTable/index_register=[20](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:20)74,tmpRegister=2064/select_last_with_five_targets (0.01s)
20
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
[21](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:21)
	panic: runtime error: invalid memory address or nil pointer dereference
[22](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:22)
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x24ae066]
[23](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:23)

[24](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:24)
goroutine 862 [running]:
[25](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:25)
testing.tRunner.func1.2({0x12ae360, 0x1513370})
[26](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:26)
	/Users/runner/hostedtoolcache/go/1.17.7/x64/src/testing/testing.go:1209 +0x24e
[27](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:27)
testing.tRunner.func1()
[28](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:28)
	/Users/runner/hostedtoolcache/go/1.17.7/x64/src/testing/testing.go:1212 +0x218
[29](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:29)
panic({0x12ae360, 0x1513370})
[30](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:30)
	/Users/runner/hostedtoolcache/go/1.17.7/x64/src/runtime/panic.go:1038 +0x215
[31](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:31)
github.com/tetratelabs/wazero/internal/wasm/jit.(*jitEnv).exec(0xc00000e078, {0x24ae000, 0x1a9, 0x1a9})
[32](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:32)
	/Users/runner/work/wazero/wazero/internal/wasm/jit/jit_test.go:122 +0x20e
[33](https://github.com/tetratelabs/wazero/runs/5415751283?check_suite_focus=true#step:5:33)
github.com/tetratelabs/wazero/internal/wasm/jit.TestAmd64Compiler_compileBrTable.func1(0x1361978, 0xc00009ad80, 0x12abe40)
```

--- a/internal/wasm/jit/engine.go
+++ b/internal/wasm/jit/engine.go
@@ -365,7 +365,7 @@ func (e *engine) Call(ctx *wasm.ModuleContext, f *wasm.FunctionInstance, params 
 		return nil, fmt.Errorf("expected %d params, but passed %d", len(paramSignature), paramCount)
 	}
 
-	vm := e.newCallEngine()
+	ce := e.newCallEngine()
 
 	// We ensure that this Call method never panics as
 	// this Call method is indirectly invoked by embedders via store.CallFunction,
@@ -378,8 +378,8 @@ func (e *engine) Call(ctx *wasm.ModuleContext, f *wasm.FunctionInstance, params 
 			}
 
 			var frames []string
-			for i := uint64(0); i < vm.globalContext.callFrameStackPointer; i++ {
-				f := vm.callFrameStack[vm.globalContext.callFrameStackPointer-1-i].compiledFunction
+			for i := uint64(0); i < ce.globalContext.callFrameStackPointer; i++ {
+				f := ce.callFrameStack[ce.globalContext.callFrameStackPointer-1-i].compiledFunction
 				frames = append(frames, fmt.Sprintf("\t%d: %s", i, f.source.Name))
 				// TODO: include DWARF symbols. See #58
 			}
@@ -398,26 +398,26 @@ func (e *engine) Call(ctx *wasm.ModuleContext, f *wasm.FunctionInstance, params 
 	}()
 
 	for _, param := range params {
-		vm.pushValue(param)
+		ce.pushValue(param)
 	}
 
-	compiled := vm.compiledFunctions[f.Index]
+	compiled := ce.compiledFunctions[f.Index]
 	if compiled == nil {
 		err = fmt.Errorf("function not compiled")
 		return
 	}
 
 	if f.FunctionKind == wasm.FunctionKindWasm {
-		vm.execWasmFunction(ctx, compiled)
+		ce.execWasmFunction(ctx, compiled)
 	} else {
-		vm.execHostFunction(f.FunctionKind, compiled.source.HostFunction, ctx)
+		ce.execHostFunction(f.FunctionKind, compiled.source.HostFunction, ctx)
 	}
 
 	// Note the top value is the tail of the results,
 	// so we assign them in reverse order.
 	results = make([]uint64, len(f.FunctionType.Type.Results))
 	for i := range results {
-		results[len(results)-1-i] = vm.popValue()
+		results[len(results)-1-i] = ce.popValue()
 	}
 	return
 }
@@ -444,17 +444,17 @@ func (e *engine) newCallEngine() *callEngine {
 	e.mux.RLock()
 	defer e.mux.RUnlock()
 
-	vm := &callEngine{
+	ce := &callEngine{
 		valueStack:        make([]uint64, initialValueStackSize),
 		callFrameStack:    make([]callFrame, initialCallFrameStackSize),
 		archContext:       newArchContext(),
 		compiledFunctions: e.compiledFunctions,
 	}
 
-	valueStackHeader := (*reflect.SliceHeader)(unsafe.Pointer(&vm.valueStack))
-	callFrameStackHeader := (*reflect.SliceHeader)(unsafe.Pointer(&vm.callFrameStack))
-	compiledFunctionsHeader := (*reflect.SliceHeader)(unsafe.Pointer(&vm.compiledFunctions))
-	vm.globalContext = globalContext{
+	valueStackHeader := (*reflect.SliceHeader)(unsafe.Pointer(&ce.valueStack))
+	callFrameStackHeader := (*reflect.SliceHeader)(unsafe.Pointer(&ce.callFrameStack))
+	compiledFunctionsHeader := (*reflect.SliceHeader)(unsafe.Pointer(&ce.compiledFunctions))
+	ce.globalContext = globalContext{
 		valueStackElement0Address:        valueStackHeader.Data,
 		valueStackLen:                    uint64(valueStackHeader.Len),
 		callFrameStackElementZeroAddress: callFrameStackHeader.Data,
@@ -462,30 +462,30 @@ func (e *engine) newCallEngine() *callEngine {
 		callFrameStackPointer:            0,
 		compiledFunctionsElement0Address: compiledFunctionsHeader.Data,
 	}
-	return vm
+	return ce
 }
 
-func (vm *callEngine) popValue() (ret uint64) {
-	vm.valueStackContext.stackPointer--
-	ret = vm.valueStack[vm.valueStackTopIndex()]
+func (ce *callEngine) popValue() (ret uint64) {
+	ce.valueStackContext.stackPointer--
+	ret = ce.valueStack[ce.valueStackTopIndex()]
 	return
 }
 
-func (vm *callEngine) pushValue(v uint64) {
-	vm.valueStack[vm.valueStackTopIndex()] = v
-	vm.valueStackContext.stackPointer++
+func (ce *callEngine) pushValue(v uint64) {
+	ce.valueStack[ce.valueStackTopIndex()] = v
+	ce.valueStackContext.stackPointer++
 }
 
-func (vm *callEngine) callFrameTop() *callFrame {
-	return &vm.callFrameStack[vm.globalContext.callFrameStackPointer-1]
+func (ce *callEngine) callFrameTop() *callFrame {
+	return &ce.callFrameStack[ce.globalContext.callFrameStackPointer-1]
 }
 
-func (vm *callEngine) callFrameAt(depth uint64) *callFrame {
-	return &vm.callFrameStack[vm.globalContext.callFrameStackPointer-1-depth]
+func (ce *callEngine) callFrameAt(depth uint64) *callFrame {
+	return &ce.callFrameStack[ce.globalContext.callFrameStackPointer-1-depth]
 }
 
-func (vm *callEngine) valueStackTopIndex() uint64 {
-	return vm.valueStackContext.stackBasePointer + vm.valueStackContext.stackPointer
+func (ce *callEngine) valueStackTopIndex() uint64 {
+	return ce.valueStackContext.stackBasePointer + ce.valueStackContext.stackPointer
 }
 
 const (
@@ -506,7 +506,7 @@ const (
 // After the execution, the result of host function is pushed onto the stack.
 //
 // ctx parameter is passed to the host function as a first argument.
-func (vm *callEngine) execHostFunction(fk wasm.FunctionKind, f *reflect.Value, ctx *wasm.ModuleContext) {
+func (ce *callEngine) execHostFunction(fk wasm.FunctionKind, f *reflect.Value, ctx *wasm.ModuleContext) {
 	// TODO: the signature won't ever change for a host function once instantiated. For this reason, we should be able
 	// to optimize below based on known possible outcomes. This includes knowledge about if it has a context param[0]
 	// and which type (if any) it returns.
@@ -521,7 +521,7 @@ func (vm *callEngine) execHostFunction(fk wasm.FunctionKind, f *reflect.Value, c
 	}
 	for i := len(in) - 1; i >= wasmParamOffset; i-- {
 		val := reflect.New(tp.In(i)).Elem()
-		raw := vm.popValue()
+		raw := ce.popValue()
 		kind := tp.In(i).Kind()
 		switch kind {
 		case reflect.Float32:
@@ -545,60 +545,60 @@ func (vm *callEngine) execHostFunction(fk wasm.FunctionKind, f *reflect.Value, c
 	for _, ret := range f.Call(in) {
 		switch ret.Kind() {
 		case reflect.Float32:
-			vm.pushValue(uint64(math.Float32bits(float32(ret.Float()))))
+			ce.pushValue(uint64(math.Float32bits(float32(ret.Float()))))
 		case reflect.Float64:
-			vm.pushValue(math.Float64bits(ret.Float()))
+			ce.pushValue(math.Float64bits(ret.Float()))
 		case reflect.Uint32, reflect.Uint64:
-			vm.pushValue(ret.Uint())
+			ce.pushValue(ret.Uint())
 		case reflect.Int32, reflect.Int64:
-			vm.pushValue(uint64(ret.Int()))
+			ce.pushValue(uint64(ret.Int()))
 		default:
 			panic("invalid return type")
 		}
 	}
 }
 
-func (vm *callEngine) execWasmFunction(ctx *wasm.ModuleContext, f *compiledFunction) {
-	vm.pushCallFrame(f)
+func (ce *callEngine) execWasmFunction(ctx *wasm.ModuleContext, f *compiledFunction) {
+	ce.pushCallFrame(f)
 
 jitentry:
 	{
-		frame := vm.callFrameTop()
+		frame := ce.callFrameTop()
 		if buildoptions.IsDebugMode {
 			fmt.Printf("callframe=%s, stackBasePointer: %d, stackPointer: %d\n",
-				frame.String(), vm.valueStackContext.stackBasePointer, vm.valueStackContext.stackPointer)
+				frame.String(), ce.valueStackContext.stackBasePointer, ce.valueStackContext.stackPointer)
 		}
 
 		// Call into the JIT code.
-		jitcall(frame.returnAddress, uintptr(unsafe.Pointer(vm)))
+		jitcall(frame.returnAddress, uintptr(unsafe.Pointer(ce)))
 
 		// Check the status code from JIT code.
-		switch status := vm.exitContext.statusCode; status {
+		switch status := ce.exitContext.statusCode; status {
 		case jitCallStatusCodeReturned:
 			// Meaning that all the function frames above the previous call frame stack pointer are executed.
 		case jitCallStatusCodeCallHostFunction:
 			// Not "callFrameTop" but take the below of peek with "callFrameAt(1)" as the top frame is for host function,
 			// but when making host function calls, we need to pass the memory instance of host function caller.
-			fn := vm.compiledFunctions[vm.exitContext.functionCallAddress]
-			callerCompiledFunction := vm.callFrameAt(1).compiledFunction
+			fn := ce.compiledFunctions[ce.exitContext.functionCallAddress]
+			callerCompiledFunction := ce.callFrameAt(1).compiledFunction
 			// A host function is invoked with the calling frame's memory, which may be different if in another module.
-			vm.execHostFunction(fn.source.FunctionKind, fn.source.HostFunction,
+			ce.execHostFunction(fn.source.FunctionKind, fn.source.HostFunction,
 				ctx.WithMemory(callerCompiledFunction.source.ModuleInstance.MemoryInstance),
 			)
 			goto jitentry
 		case jitCallStatusCodeCallBuiltInFunction:
-			switch vm.exitContext.functionCallAddress {
+			switch ce.exitContext.functionCallAddress {
 			case builtinFunctionIndexMemoryGrow:
-				callerCompiledFunction := vm.callFrameTop().compiledFunction
-				vm.builtinFunctionMemoryGrow(callerCompiledFunction.source.ModuleInstance.MemoryInstance)
+				callerCompiledFunction := ce.callFrameTop().compiledFunction
+				ce.builtinFunctionMemoryGrow(callerCompiledFunction.source.ModuleInstance.MemoryInstance)
 			case builtinFunctionIndexGrowValueStack:
-				callerCompiledFunction := vm.callFrameTop().compiledFunction
-				vm.builtinFunctionGrowValueStack(callerCompiledFunction.stackPointerCeil)
+				callerCompiledFunction := ce.callFrameTop().compiledFunction
+				ce.builtinFunctionGrowValueStack(callerCompiledFunction.stackPointerCeil)
 			case builtinFunctionIndexGrowCallFrameStack:
-				vm.builtinFunctionGrowCallFrameStack()
+				ce.builtinFunctionGrowCallFrameStack()
 			}
 			if buildoptions.IsDebugMode {
-				if vm.exitContext.functionCallAddress == builtinFunctionIndexBreakPoint {
+				if ce.exitContext.functionCallAddress == builtinFunctionIndexBreakPoint {
 					runtime.Breakpoint()
 				}
 			}
@@ -610,10 +610,10 @@ jitentry:
 }
 
 // pushInitialFrame is implemented in assembly as well, but this Go version is used BEFORE jit entry.
-func (vm *callEngine) pushCallFrame(f *compiledFunction) {
+func (ce *callEngine) pushCallFrame(f *compiledFunction) {
 	// Push the new frame to the top of stack.
-	vm.callFrameStack[vm.globalContext.callFrameStackPointer] = callFrame{returnAddress: f.codeInitialAddress, compiledFunction: f}
-	vm.globalContext.callFrameStackPointer++
+	ce.callFrameStack[ce.globalContext.callFrameStackPointer] = callFrame{returnAddress: f.codeInitialAddress, compiledFunction: f}
+	ce.globalContext.callFrameStackPointer++
 
 	// For example, if we have the following state (where "_" means no value pushed),
 	//       base            sp
@@ -628,53 +628,53 @@ func (vm *callEngine) pushCallFrame(f *compiledFunction) {
 	// [...., A, B, C, D, E, _, _ ]
 	//
 	// That maens the next stack base poitner is calculated as follows (note stack pointer is relative to base):
-	vm.valueStackContext.stackBasePointer =
-		vm.valueStackContext.stackBasePointer + vm.valueStackContext.stackPointer - uint64(len(f.source.FunctionType.Type.Params))
+	ce.valueStackContext.stackBasePointer =
+		ce.valueStackContext.stackBasePointer + ce.valueStackContext.stackPointer - uint64(len(f.source.FunctionType.Type.Params))
 }
 
-func (vm *callEngine) builtinFunctionGrowValueStack(stackPointerCeil uint64) {
+func (ce *callEngine) builtinFunctionGrowValueStack(stackPointerCeil uint64) {
 	// Extends the valueStack's length to currentLen*2+stackPointerCeil.
-	newLen := vm.globalContext.valueStackLen*2 + (stackPointerCeil)
+	newLen := ce.globalContext.valueStackLen*2 + (stackPointerCeil)
 	newStack := make([]uint64, newLen)
-	top := vm.valueStackContext.stackBasePointer + vm.valueStackContext.stackPointer
-	copy(newStack[:top], vm.valueStack[:top])
-	vm.valueStack = newStack
+	top := ce.valueStackContext.stackBasePointer + ce.valueStackContext.stackPointer
+	copy(newStack[:top], ce.valueStack[:top])
+	ce.valueStack = newStack
 
 	// Update the globalContext's fields as they become stale after the update ^^.
 	stackSliceHeader := (*reflect.SliceHeader)(unsafe.Pointer(&newStack))
-	vm.globalContext.valueStackElement0Address = stackSliceHeader.Data
-	vm.globalContext.valueStackLen = uint64(stackSliceHeader.Len)
+	ce.globalContext.valueStackElement0Address = stackSliceHeader.Data
+	ce.globalContext.valueStackLen = uint64(stackSliceHeader.Len)
 }
 
 var callStackCeiling = uint64(buildoptions.CallStackCeiling)
 
-func (vm *callEngine) builtinFunctionGrowCallFrameStack() {
-	if callStackCeiling < uint64(len(vm.callFrameStack)+1) {
+func (ce *callEngine) builtinFunctionGrowCallFrameStack() {
+	if callStackCeiling < uint64(len(ce.callFrameStack)+1) {
 		panic(wasm.ErrRuntimeCallStackOverflow)
 	}
 
 	// Double the callstack slice length.
-	newLen := uint64(vm.globalContext.callFrameStackLen) * 2
+	newLen := uint64(ce.globalContext.callFrameStackLen) * 2
 	newStack := make([]callFrame, newLen)
-	copy(newStack, vm.callFrameStack)
-	vm.callFrameStack = newStack
+	copy(newStack, ce.callFrameStack)
+	ce.callFrameStack = newStack
 
 	// Update the globalContext's fields as they become stale after the update ^^.
 	stackSliceHeader := (*reflect.SliceHeader)(unsafe.Pointer(&newStack))
-	vm.globalContext.callFrameStackLen = uint64(stackSliceHeader.Len)
-	vm.globalContext.callFrameStackElementZeroAddress = stackSliceHeader.Data
+	ce.globalContext.callFrameStackLen = uint64(stackSliceHeader.Len)
+	ce.globalContext.callFrameStackElementZeroAddress = stackSliceHeader.Data
 }
 
-func (e *callEngine) builtinFunctionMemoryGrow(mem *wasm.MemoryInstance) {
-	newPages := e.popValue()
+func (ce *callEngine) builtinFunctionMemoryGrow(mem *wasm.MemoryInstance) {
+	newPages := ce.popValue()
 
 	res := mem.Grow(uint32(newPages))
-	e.pushValue(uint64(res))
+	ce.pushValue(uint64(res))
 
 	// Update the moduleContext's fields as they become stale after the update ^^.
 	bufSliceHeader := (*reflect.SliceHeader)(unsafe.Pointer(&mem.Buffer))
-	e.moduleContext.memorySliceLen = uint64(bufSliceHeader.Len)
-	e.moduleContext.memoryElement0Address = bufSliceHeader.Data
+	ce.moduleContext.memorySliceLen = uint64(bufSliceHeader.Len)
+	ce.moduleContext.memoryElement0Address = bufSliceHeader.Data
 }
 
 func compileHostFunction(f *wasm.FunctionInstance) (*compiledFunction, error) {

--- a/internal/wasm/jit/engine_test.go
+++ b/internal/wasm/jit/engine_test.go
@@ -16,30 +16,30 @@ import (
 
 // Ensures that the offset consts do not drift when we manipulate the target structs.
 func TestVerifyOffsetValue(t *testing.T) {
-	var vm callEngine
+	var ce callEngine
 	// Offsets for callEngine.globalContext.
-	require.Equal(t, int(unsafe.Offsetof(vm.valueStackElement0Address)), callEngineGlobalContextValueStackElement0AddressOffset)
-	require.Equal(t, int(unsafe.Offsetof(vm.valueStackLen)), callEngineGlobalContextValueStackLenOffset)
-	require.Equal(t, int(unsafe.Offsetof(vm.callFrameStackElementZeroAddress)), callEngineGlobalContextCallFrameStackElement0AddressOffset)
-	require.Equal(t, int(unsafe.Offsetof(vm.callFrameStackLen)), callEngineGlobalContextCallFrameStackLenOffset)
-	require.Equal(t, int(unsafe.Offsetof(vm.callFrameStackPointer)), callEngineGlobalContextCallFrameStackPointerOffset)
-	require.Equal(t, int(unsafe.Offsetof(vm.compiledFunctionsElement0Address)), callEngineGlobalContextCompiledFunctionsElement0AddressOffset)
+	require.Equal(t, int(unsafe.Offsetof(ce.valueStackElement0Address)), callEngineGlobalContextValueStackElement0AddressOffset)
+	require.Equal(t, int(unsafe.Offsetof(ce.valueStackLen)), callEngineGlobalContextValueStackLenOffset)
+	require.Equal(t, int(unsafe.Offsetof(ce.callFrameStackElementZeroAddress)), callEngineGlobalContextCallFrameStackElement0AddressOffset)
+	require.Equal(t, int(unsafe.Offsetof(ce.callFrameStackLen)), callEngineGlobalContextCallFrameStackLenOffset)
+	require.Equal(t, int(unsafe.Offsetof(ce.callFrameStackPointer)), callEngineGlobalContextCallFrameStackPointerOffset)
+	require.Equal(t, int(unsafe.Offsetof(ce.compiledFunctionsElement0Address)), callEngineGlobalContextCompiledFunctionsElement0AddressOffset)
 
 	// Offsets for callEngine.moduleContext.
-	require.Equal(t, int(unsafe.Offsetof(vm.moduleInstanceAddress)), callEngineModuleContextModuleInstanceAddressOffset)
-	require.Equal(t, int(unsafe.Offsetof(vm.globalElement0Address)), callEngineModuleContextGlobalElement0AddressOffset)
-	require.Equal(t, int(unsafe.Offsetof(vm.memoryElement0Address)), callEngineModuleContextMemoryElement0AddressOffset)
-	require.Equal(t, int(unsafe.Offsetof(vm.memorySliceLen)), callEngineModuleContextMemorySliceLenOffset)
-	require.Equal(t, int(unsafe.Offsetof(vm.tableElement0Address)), callEngineModuleContextTableElement0AddressOffset)
-	require.Equal(t, int(unsafe.Offsetof(vm.tableSliceLen)), callEngineModuleContextTableSliceLenOffset)
+	require.Equal(t, int(unsafe.Offsetof(ce.moduleInstanceAddress)), callEngineModuleContextModuleInstanceAddressOffset)
+	require.Equal(t, int(unsafe.Offsetof(ce.globalElement0Address)), callEngineModuleContextGlobalElement0AddressOffset)
+	require.Equal(t, int(unsafe.Offsetof(ce.memoryElement0Address)), callEngineModuleContextMemoryElement0AddressOffset)
+	require.Equal(t, int(unsafe.Offsetof(ce.memorySliceLen)), callEngineModuleContextMemorySliceLenOffset)
+	require.Equal(t, int(unsafe.Offsetof(ce.tableElement0Address)), callEngineModuleContextTableElement0AddressOffset)
+	require.Equal(t, int(unsafe.Offsetof(ce.tableSliceLen)), callEngineModuleContextTableSliceLenOffset)
 
 	// Offsets for callEngine.valueStackContext
-	require.Equal(t, int(unsafe.Offsetof(vm.stackPointer)), callEngineValueStackContextStackPointerOffset)
-	require.Equal(t, int(unsafe.Offsetof(vm.stackBasePointer)), callEngineValueStackContextStackBasePointerOffset)
+	require.Equal(t, int(unsafe.Offsetof(ce.stackPointer)), callEngineValueStackContextStackPointerOffset)
+	require.Equal(t, int(unsafe.Offsetof(ce.stackBasePointer)), callEngineValueStackContextStackBasePointerOffset)
 
 	// Offsets for callEngine.exitContext.
-	require.Equal(t, int(unsafe.Offsetof(vm.statusCode)), callEngineExitContextJITCallStatusCodeOffset)
-	require.Equal(t, int(unsafe.Offsetof(vm.functionCallAddress)), callEngineExitContextFunctionCallAddressOffset)
+	require.Equal(t, int(unsafe.Offsetof(ce.statusCode)), callEngineExitContextJITCallStatusCodeOffset)
+	require.Equal(t, int(unsafe.Offsetof(ce.functionCallAddress)), callEngineExitContextFunctionCallAddressOffset)
 
 	// Size and offsets for callFrame.
 	var frame callFrame

--- a/internal/wasm/jit/jit_amd64.go
+++ b/internal/wasm/jit/jit_amd64.go
@@ -76,10 +76,10 @@ func init() {
 }
 
 // jitcall is implemented in jit_amd64.s as a Go Assembler function.
-// This is used by callEngine.exec and the entrypoint to enter the JITed native code.
+// This is used by callEngine.execWasmFunction and the entrypoint to enter the JITed native code.
 // codeSegment is the pointer to the initial instruction of the compiled native code.
-// vm is the pointer to the callEngine as uintptr.
-func jitcall(codeSegment, vm uintptr)
+// ce is the pointer to the *callEngine as uintptr.
+func jitcall(codeSegment, ce uintptr)
 
 // archContext is embedded in callEngine in order to store architecture-specific data.
 // For amd64, this is empty.

--- a/internal/wasm/jit/jit_amd64.s
+++ b/internal/wasm/jit/jit_amd64.s
@@ -3,8 +3,8 @@
 #include "funcdata.h"
 #include "textflag.h"
 
-// jitcall(codeSegment, callEngine)
+// jitcall(codeSegment, ce)
 TEXT ·jitcall(SB),NOSPLIT|NOFRAME,$0-16
-        MOVQ codeSegment+0(FP),AX      // Load the address of native code.
-        MOVQ callEngine+8(FP),R13  // Load the address of callEngine.
-        JMP AX                         // Jump to native code.
+        MOVQ codeSegment+0(FP),AX  // Load the address of native code.
+        MOVQ ce+8(FP),R13          // Load the address of *callEngine.
+        JMP AX                     // Jump to native code.

--- a/internal/wasm/jit/jit_amd64_test.go
+++ b/internal/wasm/jit/jit_amd64_test.go
@@ -158,7 +158,7 @@ func TestAmd64Compiler_returnFunction(t *testing.T) {
 	t.Run("deep call stack", func(t *testing.T) {
 		env := newJITEnvironment()
 		engine := env.engine()
-		vm := env.callEngine()
+		ce := env.callEngine()
 
 		// Push the call frames.
 		const callFrameNums = 10
@@ -193,8 +193,8 @@ func TestAmd64Compiler_returnFunction(t *testing.T) {
 				returnStackBasePointer: uint64(funcaddr) * 5,
 				compiledFunction:       compiledFunction,
 			}
-			vm.callFrameStack[vm.globalContext.callFrameStackPointer] = frame
-			vm.globalContext.callFrameStackPointer++
+			ce.callFrameStack[ce.globalContext.callFrameStackPointer] = frame
+			ce.globalContext.callFrameStackPointer++
 
 			stackPointerToExpectedValue[frame.returnStackBasePointer] = expValue
 		}
@@ -202,7 +202,7 @@ func TestAmd64Compiler_returnFunction(t *testing.T) {
 		require.Equal(t, uint64(callFrameNums), env.callFrameStackPointer())
 
 		// Run codes.
-		env.exec(vm.callFrameTop().compiledFunction.codeSegment)
+		env.exec(ce.callFrameTop().compiledFunction.codeSegment)
 
 		// Check the exit status.
 		require.Equal(t, jitCallStatusCodeReturned, env.jitStatus())
@@ -284,7 +284,7 @@ func TestAmd64Compiler_initializeModuleContext(t *testing.T) {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
 			env := newJITEnvironment()
-			vm := env.callEngine()
+			ce := env.callEngine()
 			compiler := env.requireNewCompiler(t)
 			compiler.initializeReservedStackBasePointer()
 			compiler.f.ModuleInstance = tc.moduleInstance
@@ -310,25 +310,25 @@ func TestAmd64Compiler_initializeModuleContext(t *testing.T) {
 
 			// Check if the fields of callEngine.moduleContext are updated.
 			bufSliceHeader := (*reflect.SliceHeader)(unsafe.Pointer(&tc.moduleInstance.Globals))
-			require.Equal(t, bufSliceHeader.Data, vm.moduleContext.globalElement0Address)
+			require.Equal(t, bufSliceHeader.Data, ce.moduleContext.globalElement0Address)
 
 			if tc.moduleInstance.MemoryInstance != nil {
 				bufSliceHeader := (*reflect.SliceHeader)(unsafe.Pointer(&tc.moduleInstance.MemoryInstance.Buffer))
-				require.Equal(t, uint64(bufSliceHeader.Len), vm.moduleContext.memorySliceLen)
-				require.Equal(t, bufSliceHeader.Data, vm.moduleContext.memoryElement0Address)
+				require.Equal(t, uint64(bufSliceHeader.Len), ce.moduleContext.memorySliceLen)
+				require.Equal(t, bufSliceHeader.Data, ce.moduleContext.memoryElement0Address)
 			}
 
 			if tc.moduleInstance.TableInstance != nil {
 				tableHeader := (*reflect.SliceHeader)(unsafe.Pointer(&tc.moduleInstance.TableInstance.Table))
-				require.Equal(t, uint64(tableHeader.Len), vm.moduleContext.tableSliceLen)
-				require.Equal(t, tableHeader.Data, vm.moduleContext.tableElement0Address)
+				require.Equal(t, uint64(tableHeader.Len), ce.moduleContext.tableSliceLen)
+				require.Equal(t, tableHeader.Data, ce.moduleContext.tableElement0Address)
 			}
 		})
 	}
 }
 
 func TestAmd64Compiler_compileBrTable(t *testing.T) {
-	requireRunAndExpectedValueReturned := func(t *testing.T, c *amd64Compiler, expValue uint32) {
+	requireRunAndExpectedValueReturned := func(t *testing.T, env *jitEnv, c *amd64Compiler, expValue uint32) {
 		// Emit code for each label which returns the frame ID.
 		for returnValue := uint32(0); returnValue < 10; returnValue++ {
 			label := &wazeroir.Label{Kind: wazeroir.LabelKindHeader, FrameID: returnValue}
@@ -345,7 +345,6 @@ func TestAmd64Compiler_compileBrTable(t *testing.T) {
 		require.NoError(t, err)
 
 		// Run codes
-		env := newJITEnvironment()
 		env.exec(code)
 
 		// Check the returned value.
@@ -503,7 +502,7 @@ func TestAmd64Compiler_compileBrTable(t *testing.T) {
 						require.NotContains(t, compiler.locationStack.usedRegisters, indexReg)
 						require.NotContains(t, compiler.locationStack.usedRegisters, tmpReg)
 
-						requireRunAndExpectedValueReturned(t, compiler, tc.expectedValue)
+						requireRunAndExpectedValueReturned(t, env, compiler, tc.expectedValue)
 					})
 				}
 			})

--- a/internal/wasm/jit/jit_arm64.go
+++ b/internal/wasm/jit/jit_arm64.go
@@ -37,7 +37,7 @@ func newArchContext() archContext {
 type archContext struct {
 	// jitCallReturnAddress holds the absolute return address for jitcall.
 	// The value is set whenever jitcall is executed and done in jit_arm64.s
-	// Native code can return back to the vm.exec's main loop back by
+	// Native code can return back to the ce.execWasmFunction's main loop back by
 	// executing "ret" instruction with this value. See arm64Compiler.exit.
 	// Note: this is only used by JIT code so mark this as nolint.
 	jitCallReturnAddress uint64 //nolint
@@ -67,8 +67,8 @@ const (
 // jitcall is implemented in jit_arm64.s as a Go Assembler function.
 // This is used by callEngine.execWasmFunction and the entrypoint to enter the JITed native code.
 // codeSegment is the pointer to the initial instruction of the compiled native code.
-// vm is "*callEngine" as uintptr.
-func jitcall(codeSegment, vm uintptr)
+// ce is "*callEngine" as uintptr.
+func jitcall(codeSegment, ce uintptr)
 
 // golang-asm is not goroutine-safe so we take lock until we complete the compilation.
 // TODO: delete after https://github.com/tetratelabs/wazero/issues/233
@@ -446,21 +446,21 @@ func (c *arm64Compiler) compileMaybeGrowValueStack() error {
 	}
 	tmpX, tmpY := tmpRegs[0], tmpRegs[1]
 
-	// "tmpX = len(vm.valueStack)"
+	// "tmpX = len(ce.valueStack)"
 	c.compileMemoryToRegisterInstruction(
 		arm64.AMOVD,
 		reservedRegisterForCallEngine, callEngineGlobalContextValueStackLenOffset,
 		tmpX,
 	)
 
-	// "tmpY = vm.stackBasePointer"
+	// "tmpY = ce.stackBasePointer"
 	c.compileMemoryToRegisterInstruction(
 		arm64.AMOVD,
 		reservedRegisterForCallEngine, callEngineValueStackContextStackBasePointerOffset,
 		tmpY,
 	)
 
-	// "tmpX = tmpX - tmpY", in other words "tmpX = len(vm.valueStack) - vm.stackBasePointer"
+	// "tmpX = tmpX - tmpY", in other words "tmpX = len(ce.valueStack) - ce.stackBasePointer"
 	c.compileRegisterToRegisterInstruction(
 		arm64.ASUB,
 		tmpY,
@@ -477,7 +477,7 @@ func (c *arm64Compiler) compileMaybeGrowValueStack() error {
 	// so we layzily resolve the value later.
 	c.onStackPointerCeilDeterminedCallBack = func(stackPointerCeil uint64) { loadStackPointerCeil.From.Offset = int64(stackPointerCeil) }
 
-	// Compare tmpX (len(vm.valueStack) - vm.stackBasePointer) and tmpY (vm.stackPointerCeil)
+	// Compare tmpX (len(ce.valueStack) - ce.stackBasePointer) and tmpY (ce.stackPointerCeil)
 	c.compileTwoRegistersToNoneInstruction(arm64.ACMP, tmpX, tmpY)
 
 	// If ceil > valueStackLen - stack base pointer, we need to grow the stack by calling builtin Go function.
@@ -529,7 +529,7 @@ func (c *arm64Compiler) compileReturnFunction() error {
 
 	// First, we have to calculate the caller callFrame's absolute address to aquire the return address.
 	//
-	// "tmpReg = &vm.callFrameStack[0]"
+	// "tmpReg = &ce.callFrameStack[0]"
 	c.compileMemoryToRegisterInstruction(arm64.AMOVD,
 		reservedRegisterForCallEngine, callEngineGlobalContextCallFrameStackElement0AddressOffset,
 		tmpReg,
@@ -546,7 +546,7 @@ func (c *arm64Compiler) compileReturnFunction() error {
 	//      [......., ra.caller, rb.caller, rc.caller, _, ra.current, rb.current, rc.current, _, ...]  <- call frame stack's data region (somewhere in the memory)
 	//                                                  ^
 	//                               callFrameStackTopAddressRegister
-	//                   (absolute address of &callFrameStack[vm.callFrameStackPointer])
+	//                   (absolute address of &callFrameStack[ce.callFrameStackPointer])
 	//
 	// where:
 	//      ra.* = callFrame.returnAddress
@@ -555,10 +555,10 @@ func (c *arm64Compiler) compileReturnFunction() error {
 	//      _  = callFrame's padding (see comment on callFrame._ field.)
 	//
 	// What we have to do in the following is that
-	//   1) Set vm.valueStackContext.stackBasePointer to the value on "rb.caller".
+	//   1) Set ce.valueStackContext.stackBasePointer to the value on "rb.caller".
 	//   2) Jump into the address of "ra.caller".
 
-	// 1) Set vm.valueStackContext.stackBasePointer to the value on "rb.caller".
+	// 1) Set ce.valueStackContext.stackBasePointer to the value on "rb.caller".
 	c.compileMemoryToRegisterInstruction(arm64.AMOVD,
 		// "rb.caller" is below the top address.
 		callFrameStackTopAddressRegister, -(callFrameDataSize - callFrameReturnStackBasePointerOffset),
@@ -578,9 +578,9 @@ func (c *arm64Compiler) compileReturnFunction() error {
 	return nil
 }
 
-// compileExitFromNativeCode adds instructions to give the control back to vm.exec with the given status code.
+// compileExitFromNativeCode adds instructions to give the control back to ce.exec with the given status code.
 func (c *arm64Compiler) compileExitFromNativeCode(status jitCallStatusCode) error {
-	// Write the current stack pointer to the vm.stackPointer.
+	// Write the current stack pointer to the ce.stackPointer.
 	c.compileConstToRegisterInstruction(arm64.AMOVD, int64(c.locationStack.sp), reservedRegisterForTemporary)
 	c.compileRegisterToMemoryInstruction(arm64.AMOVD, reservedRegisterForTemporary, reservedRegisterForCallEngine,
 		callEngineValueStackContextStackPointerOffset)
@@ -594,7 +594,7 @@ func (c *arm64Compiler) compileExitFromNativeCode(status jitCallStatusCode) erro
 	}
 
 	// The return address to the Go code is stored in archContext.jitReturnAddress which
-	// is embedded in vm. We load the value to the tmpRegister, and then
+	// is embedded in ce. We load the value to the tmpRegister, and then
 	// invoke RET with that register.
 	c.compileMemoryToRegisterInstruction(arm64.AMOVD, reservedRegisterForCallEngine, callEngineArchContextJITCallReturnAddressOffset, reservedRegisterForTemporary)
 
@@ -1111,17 +1111,17 @@ func (c *arm64Compiler) compileCallImpl(addr wasm.FunctionIndex, indexRegister i
 
 	// First, we have to check if we need to grow the callFrame stack.
 	//
-	// "callFrameStackPointerRegister = vm.callFrameStackPointer"
+	// "callFrameStackPointerRegister = ce.callFrameStackPointer"
 	c.compileMemoryToRegisterInstruction(arm64.AMOVD,
 		reservedRegisterForCallEngine, callEngineGlobalContextCallFrameStackPointerOffset,
 		callFrameStackPointerRegister)
-	// "tmp = len(vm.callFrameStack)"
+	// "tmp = len(ce.callFrameStack)"
 	c.compileMemoryToRegisterInstruction(
 		arm64.AMOVD,
 		reservedRegisterForCallEngine, callEngineGlobalContextCallFrameStackLenOffset,
 		tmp,
 	)
-	// Compare tmp(len(vm.callFrameStack)) with callFrameStackPointerRegister(vm.callFrameStackPointer).
+	// Compare tmp(len(ce.callFrameStack)) with callFrameStackPointerRegister(ce.callFrameStackPointer).
 	c.compileTwoRegistersToNoneInstruction(arm64.ACMP, tmp, callFrameStackPointerRegister)
 	brIfCallFrameStackOK := c.compilelBranchInstruction(arm64.ABNE)
 
@@ -1153,8 +1153,8 @@ func (c *arm64Compiler) compileCallImpl(addr wasm.FunctionIndex, indexRegister i
 		}
 	}
 
-	// On the function return, we again have to set vm.callFrameStackPointer into callFrameStackPointerRegister.
-	// "callFrameStackPointerRegister = vm.callFrameStackPointer"
+	// On the function return, we again have to set ce.callFrameStackPointer into callFrameStackPointerRegister.
+	// "callFrameStackPointerRegister = ce.callFrameStackPointer"
 	c.compileMemoryToRegisterInstruction(arm64.AMOVD,
 		reservedRegisterForCallEngine, callEngineGlobalContextCallFrameStackPointerOffset,
 		callFrameStackPointerRegister)
@@ -1168,7 +1168,7 @@ func (c *arm64Compiler) compileCallImpl(addr wasm.FunctionIndex, indexRegister i
 	//    [..., ra.current, rb.current, rc.current, _, ra.next, rb.next, rc.next, ...]  <- call frame stack's data region (somewhere in the memory)
 	//                                               ^
 	//                              callFrameStackTopAddressRegister
-	//                (absolute address of &callFrame[vm.callFrameStackPointer]])
+	//                (absolute address of &callFrame[ce.callFrameStackPointer]])
 	//
 	// where:
 	//      ra.* = callFrame.returnAddress
@@ -1180,7 +1180,7 @@ func (c *arm64Compiler) compileCallImpl(addr wasm.FunctionIndex, indexRegister i
 	//
 	// What we have to do in the following is that
 	//   1) Set rb.current so that we can return back to this function properly.
-	//   2) Set vm.valueStackContext.stackBasePointer for the next function.
+	//   2) Set ce.valueStackContext.stackBasePointer for the next function.
 	//   3) Set rc.next to specify which function is executed on the current call frame (needs to make Go function calls).
 	//   4) Set ra.current so that we can return back to this function properly.
 
@@ -1193,11 +1193,11 @@ func (c *arm64Compiler) compileCallImpl(addr wasm.FunctionIndex, indexRegister i
 		// "rb.current" is BELOW the top address. See the above example for detail.
 		callFrameStackTopAddressRegister, -(callFrameDataSize - callFrameReturnStackBasePointerOffset))
 
-	// 2) Set vm.valueStackContext.stackBasePointer for the next function.
+	// 2) Set ce.valueStackContext.stackBasePointer for the next function.
 	//
 	// At this point, oldStackBasePointer holds the old stack base pointer. We could get the new frame's
 	// stack base pointer by "old stack base pointer + old stack pointer - # of function params"
-	// See the comments in vm.pushCallFrame which does exactly the same calculation in Go.
+	// See the comments in ce.pushCallFrame which does exactly the same calculation in Go.
 	if offset := int64(c.locationStack.sp) - int64(len(functype.Params)); offset > 0 {
 		c.compileConstToRegisterInstruction(arm64.AADD, offset, oldStackBasePointer)
 		c.compileRegisterToMemoryInstruction(arm64.AMOVD,
@@ -1207,13 +1207,13 @@ func (c *arm64Compiler) compileCallImpl(addr wasm.FunctionIndex, indexRegister i
 
 	// 3) Set rc.next to specify which function is executed on the current call frame.
 	//
-	// First, we read the address of the first item of vm.compiledFunctions slice (= &vm.compiledFunctions[0])
+	// First, we read the address of the first item of ce.compiledFunctions slice (= &ce.compiledFunctions[0])
 	// into tmp.
 	c.compileMemoryToRegisterInstruction(arm64.AMOVD,
 		reservedRegisterForCallEngine, callEngineGlobalContextCompiledFunctionsElement0AddressOffset,
 		tmp)
 
-	// Next, read the index of the target function (= &vm.compiledFunctions[offset])
+	// Next, read the index of the target function (= &ce.compiledFunctions[offset])
 	// into compiledFunctionIndexRegister.
 	if isNilRegister(indexRegister) {
 		c.compileMemoryToRegisterInstruction(
@@ -1294,9 +1294,9 @@ func (c *arm64Compiler) compileCallImpl(addr wasm.FunctionIndex, indexRegister i
 }
 
 // compileCalcCallFrameStackTopAddress adds instructions to set the absolute address of
-// vm.callFrameStack[callFrameStackPointerRegister] into destinationRegister.
+// ce.callFrameStack[callFrameStackPointerRegister] into destinationRegister.
 func (c *arm64Compiler) compileCalcCallFrameStackTopAddress(callFrameStackPointerRegister, destinationRegister int16) {
-	// "destinationRegister = &vm.callFrameStack[0]"
+	// "destinationRegister = &ce.callFrameStack[0]"
 	c.compileMemoryToRegisterInstruction(arm64.AMOVD,
 		reservedRegisterForCallEngine, callEngineGlobalContextCallFrameStackElement0AddressOffset,
 		destinationRegister)
@@ -2346,7 +2346,7 @@ func (c *arm64Compiler) compileCopysign(o *wazeroir.OperationCopysign) error {
 	//    fmov    d2, x0
 	//    vbit     v0.8b, v1.8b, v2.8b
 	//
-	// "mov freg, -9223372036854775808 (stored at vm.minimum64BitSignedInt)"
+	// "mov freg, -9223372036854775808 (stored at ce.minimum64BitSignedInt)"
 	c.compileMemoryToRegisterInstruction(
 		fmov,
 		reservedRegisterForCallEngine, minValueOffsetInVM,
@@ -3040,7 +3040,7 @@ func (c *arm64Compiler) compileCallGoFunction(jitStatus jitCallStatusCode, addr 
 	tmp, currentCallFrameStackPointerRegister, currentCallFrameTopAddressRegister, returnAddressRegister :=
 		freeRegs[0], freeRegs[1], freeRegs[2], freeRegs[3]
 
-	// Set the target function address to vm.functionCallAddress
+	// Set the target function address to ce.functionCallAddress
 	// "tmp = $addr"
 	c.compileConstToRegisterInstruction(
 		arm64.AMOVD,
@@ -3048,31 +3048,31 @@ func (c *arm64Compiler) compileCallGoFunction(jitStatus jitCallStatusCode, addr 
 		tmp,
 	)
 	// "[reservedRegisterForCallEngine + callEngineExitContextFunctionCallAddressOffset] = tmp"
-	// In other words, "vm.functionCallAddress = tmp (== $addr)"
+	// In other words, "ce.functionCallAddress = tmp (== $addr)"
 	c.compileRegisterToMemoryInstruction(
 		arm64.AMOVD,
 		tmp,
 		reservedRegisterForCallEngine, callEngineExitContextFunctionCallAddressOffset,
 	)
 
-	// Next, we have to set the return address into callFrameStack[vm.callFrameStackPointer-1].returnAddress.
+	// Next, we have to set the return address into callFrameStack[ce.callFrameStackPointer-1].returnAddress.
 	//
-	// "currentCallFrameStackPointerRegister = vm.callFrameStackPointer"
+	// "currentCallFrameStackPointerRegister = ce.callFrameStackPointer"
 	c.compileMemoryToRegisterInstruction(arm64.AMOVD,
 		reservedRegisterForCallEngine, callEngineGlobalContextCallFrameStackPointerOffset,
 		currentCallFrameStackPointerRegister)
 
-	// Set the address of callFrameStack[vm.callFrameStackPointer] into currentCallFrameTopAddressRegister.
+	// Set the address of callFrameStack[ce.callFrameStackPointer] into currentCallFrameTopAddressRegister.
 	c.compileCalcCallFrameStackTopAddress(currentCallFrameStackPointerRegister, currentCallFrameTopAddressRegister)
 
 	// Set the return address (after RET in c.exit below) into returnAddressRegister.
 	c.compileReadInstructionAddress(obj.ARET, returnAddressRegister)
 
-	// Write returnAddressRegister into callFrameStack[vm.callFrameStackPointer-1].returnAddress.
+	// Write returnAddressRegister into callFrameStack[ce.callFrameStackPointer-1].returnAddress.
 	c.compileRegisterToMemoryInstruction(
 		arm64.AMOVD,
 		returnAddressRegister,
-		// callFrameStack[vm.callFrameStackPointer-1] is below of currentCallFrameTopAddressRegister.
+		// callFrameStack[ce.callFrameStackPointer-1] is below of currentCallFrameTopAddressRegister.
 		currentCallFrameTopAddressRegister, -(callFrameDataSize - callFrameReturnAddressOffset),
 	)
 
@@ -3341,7 +3341,7 @@ func (c *arm64Compiler) compileReservedStackBasePointerRegisterInitialization() 
 		reservedRegisterForCallEngine, callEngineGlobalContextValueStackElement0AddressOffset,
 		reservedRegisterForStackBasePointerAddress)
 
-	// Next we move the base pointer (vm.stackBasePointer) to reservedRegisterForTemporary.
+	// Next we move the base pointer (ce.stackBasePointer) to reservedRegisterForTemporary.
 	c.compileMemoryToRegisterInstruction(arm64.AMOVD,
 		reservedRegisterForCallEngine, callEngineValueStackContextStackBasePointerOffset,
 		reservedRegisterForTemporary)
@@ -3356,7 +3356,7 @@ func (c *arm64Compiler) compileReservedStackBasePointerRegisterInitialization() 
 
 func (c *arm64Compiler) compileReservedMemoryRegisterInitialization() {
 	if c.f.ModuleInstance.MemoryInstance != nil {
-		// "reservedRegisterForMemory = vm.MemoryElement0Address"
+		// "reservedRegisterForMemory = ce.MemoryElement0Address"
 		c.compileMemoryToRegisterInstruction(
 			arm64.AMOVD,
 			reservedRegisterForCallEngine, callEngineModuleContextMemoryElement0AddressOffset,
@@ -3365,8 +3365,8 @@ func (c *arm64Compiler) compileReservedMemoryRegisterInitialization() {
 	}
 }
 
-// compileModuleContextInitialization adds instructions to initialize vm.ModuleContext's fields based on
-// vm.ModuleContext.ModuleInstanceAddress.
+// compileModuleContextInitialization adds instructions to initialize ce.ModuleContext's fields based on
+// ce.ModuleContext.ModuleInstanceAddress.
 // This is called in two cases: in function preamble, and on the return from (non-Go) function calls.
 func (c *arm64Compiler) compileModuleContextInitialization() error {
 	regs, found := c.locationStack.takeFreeRegisters(generalPurposeRegisterTypeInt, 3)
@@ -3382,7 +3382,7 @@ func (c *arm64Compiler) compileModuleContextInitialization() error {
 	// Note: this should be modified to support Clone() functionality per #179.
 	c.compileConstToRegisterInstruction(arm64.AMOVD, int64(uintptr(unsafe.Pointer(c.f.ModuleInstance))), moduleInstanceAddressRegister)
 
-	// "tmpX = vm.ModuleInstanceAddress"
+	// "tmpX = ce.ModuleInstanceAddress"
 	c.compileMemoryToRegisterInstruction(arm64.AMOVD, reservedRegisterForCallEngine, callEngineModuleContextModuleInstanceAddressOffset, tmpX)
 
 	// If the module instance address stays the same, we could skip the entire code below.
@@ -3391,11 +3391,11 @@ func (c *arm64Compiler) compileModuleContextInitialization() error {
 	c.addInstruction(brIfModuleUnchanged)
 
 	// Otherwise, we have to update the following fields:
-	// * vm.moduleContext.globalElement0Address
-	// * vm.moduleContext.memoryElement0Address
-	// * vm.moduleContext.memorySliceLen
-	// * vm.moduleContext.tableElement0Address
-	// * vm.moduleContext.tableSliceLen
+	// * ce.moduleContext.globalElement0Address
+	// * ce.moduleContext.memoryElement0Address
+	// * ce.moduleContext.memorySliceLen
+	// * ce.moduleContext.tableElement0Address
+	// * ce.moduleContext.tableSliceLen
 
 	// Update globalElement0Address.
 	//
@@ -3409,7 +3409,7 @@ func (c *arm64Compiler) compileModuleContextInitialization() error {
 			tmpX,
 		)
 
-		// "vm.GlobalElement0Address = tmpX (== &moduleInstance.Globals[0])"
+		// "ce.GlobalElement0Address = tmpX (== &moduleInstance.Globals[0])"
 		c.compileRegisterToMemoryInstruction(
 			arm64.AMOVD, tmpX,
 			reservedRegisterForCallEngine, callEngineModuleContextGlobalElement0AddressOffset,
@@ -3429,7 +3429,7 @@ func (c *arm64Compiler) compileModuleContextInitialization() error {
 			tmpX,
 		)
 
-		// First, we write the memory length into vm.MemorySliceLen.
+		// First, we write the memory length into ce.MemorySliceLen.
 		//
 		// "tmpY = [tmpX + memoryInstanceBufferLenOffset] (== len(memory.Buffer))"
 		c.compileMemoryToRegisterInstruction(
@@ -3437,14 +3437,14 @@ func (c *arm64Compiler) compileModuleContextInitialization() error {
 			tmpX, memoryInstanceBufferLenOffset,
 			tmpY,
 		)
-		// "vm.MemorySliceLen = tmpY".
+		// "ce.MemorySliceLen = tmpY".
 		c.compileRegisterToMemoryInstruction(
 			arm64.AMOVD,
 			tmpY,
 			reservedRegisterForCallEngine, callEngineModuleContextMemorySliceLenOffset,
 		)
 
-		// Next write vm.memoryElement0Address.
+		// Next write ce.memoryElement0Address.
 		//
 		// "tmpY = *tmpX (== &memory.Buffer[0])"
 		c.compileMemoryToRegisterInstruction(
@@ -3452,7 +3452,7 @@ func (c *arm64Compiler) compileModuleContextInitialization() error {
 			tmpX, memoryInstanceBufferOffset,
 			tmpY,
 		)
-		// "vm.memoryElement0Address = tmpY".
+		// "ce.memoryElement0Address = tmpY".
 		c.compileRegisterToMemoryInstruction(
 			arm64.AMOVD,
 			tmpY,
@@ -3473,28 +3473,28 @@ func (c *arm64Compiler) compileModuleContextInitialization() error {
 			tmpX,
 		)
 
-		// Update vm.tableElement0Address.
+		// Update ce.tableElement0Address.
 		// "tmpY = &tables[0].Table[0]"
 		c.compileMemoryToRegisterInstruction(
 			arm64.AMOVD,
 			tmpX, tableInstanceTableOffset,
 			tmpY,
 		)
-		// "vm.tableElement0Address = tmpY".
+		// "ce.tableElement0Address = tmpY".
 		c.compileRegisterToMemoryInstruction(
 			arm64.AMOVD,
 			tmpY,
 			reservedRegisterForCallEngine, callEngineModuleContextTableElement0AddressOffset,
 		)
 
-		// Update vm.tableSliceLen.
+		// Update ce.tableSliceLen.
 		// "tmpY = len(tables[0].Table)"
 		c.compileMemoryToRegisterInstruction(
 			arm64.AMOVD,
 			tmpX, tableInstanceTableLenOffset,
 			tmpY,
 		)
-		// "vm.tableSliceLen = tmpY".
+		// "ce.tableSliceLen = tmpY".
 		c.compileRegisterToMemoryInstruction(
 			arm64.AMOVD,
 			tmpY,

--- a/internal/wasm/jit/jit_arm64.s
+++ b/internal/wasm/jit/jit_arm64.s
@@ -3,12 +3,12 @@
 #include "funcdata.h"
 #include "textflag.h"
 
-// jitcall(codeSegment, engine)
+// jitcall(codeSegment, ce)
 TEXT ·jitcall(SB),NOSPLIT|NOFRAME,$0-16
         // Load the address of native code.
         MOVD codeSegment+0(FP),R1
-        // Load the address of engine.
-        MOVD engine+8(FP),R0
+        // Load the address of *callEngine.
+        MOVD ce+8(FP),R0
         // In arm64, return address is stored in R30 after jumping into the code.
         // We save the return address value into archContext.jitReturnAddress in Engine.
         // Note that the const 128 drifts after editting Engine or archContext struct. See TestArchContextOffsetInEngine.

--- a/internal/wasm/jit/jit_arm64_test.go
+++ b/internal/wasm/jit/jit_arm64_test.go
@@ -87,7 +87,7 @@ func TestArm64Compiler_returnFunction(t *testing.T) {
 	t.Run("deep call stack", func(t *testing.T) {
 		env := newJITEnvironment()
 		engine := env.engine()
-		vm := env.callEngine()
+		ce := env.callEngine()
 
 		// Push the call frames.
 		const callFrameNums = 10
@@ -125,8 +125,8 @@ func TestArm64Compiler_returnFunction(t *testing.T) {
 					returnStackBasePointer: uint64(funcaddr) * 5,
 					compiledFunction:       compiledFunction,
 				}
-				vm.callFrameStack[vm.globalContext.callFrameStackPointer] = frame
-				vm.globalContext.callFrameStackPointer++
+				ce.callFrameStack[ce.globalContext.callFrameStackPointer] = frame
+				ce.globalContext.callFrameStackPointer++
 				stackPointerToExpectedValue[frame.returnStackBasePointer] = expValue
 			})
 		}
@@ -134,7 +134,7 @@ func TestArm64Compiler_returnFunction(t *testing.T) {
 		require.Equal(t, uint64(callFrameNums), env.callFrameStackPointer())
 
 		// Run code from the top frame.
-		env.exec(vm.callFrameTop().compiledFunction.codeSegment)
+		env.exec(ce.callFrameTop().compiledFunction.codeSegment)
 
 		// Check the exit status and the values on stack.
 		require.Equal(t, jitCallStatusCodeReturned, env.jitStatus())
@@ -1051,7 +1051,7 @@ func TestArm64Compiler_compilePick(t *testing.T) {
 
 	for _, tc := range []struct {
 		name                                      string
-		pickTargetSetupFunc                       func(compiler *arm64Compiler, vm *callEngine) error
+		pickTargetSetupFunc                       func(compiler *arm64Compiler, ce *callEngine) error
 		isPickTargetFloat, isPickTargetOnRegister bool
 	}{
 		{
@@ -1072,10 +1072,10 @@ func TestArm64Compiler_compilePick(t *testing.T) {
 		},
 		{
 			name: "float on stack",
-			pickTargetSetupFunc: func(compiler *arm64Compiler, vm *callEngine) error {
+			pickTargetSetupFunc: func(compiler *arm64Compiler, ce *callEngine) error {
 				pickTargetLocation := compiler.locationStack.pushValueLocationOnStack()
 				pickTargetLocation.setRegisterType(generalPurposeRegisterTypeFloat)
-				vm.valueStack[pickTargetLocation.stackPointer] = pickTargetValue
+				ce.valueStack[pickTargetLocation.stackPointer] = pickTargetValue
 				return nil
 			},
 			isPickTargetFloat:      true,
@@ -1083,10 +1083,10 @@ func TestArm64Compiler_compilePick(t *testing.T) {
 		},
 		{
 			name: "int on stack",
-			pickTargetSetupFunc: func(compiler *arm64Compiler, vm *callEngine) error {
+			pickTargetSetupFunc: func(compiler *arm64Compiler, ce *callEngine) error {
 				pickTargetLocation := compiler.locationStack.pushValueLocationOnStack()
 				pickTargetLocation.setRegisterType(generalPurposeRegisterTypeInt)
-				vm.valueStack[pickTargetLocation.stackPointer] = pickTargetValue
+				ce.valueStack[pickTargetLocation.stackPointer] = pickTargetValue
 				return nil
 			},
 			isPickTargetFloat:      false,
@@ -1235,7 +1235,7 @@ func TestArm64Compiler_compileDrop(t *testing.T) {
 		liveTotal := liveAboveDropStartNum + liveBelowDropEndNum
 
 		env := newJITEnvironment()
-		vm := env.callEngine()
+		ce := env.callEngine()
 		compiler := env.requireNewCompiler(t)
 
 		err := compiler.compilePreamble()
@@ -1244,7 +1244,7 @@ func TestArm64Compiler_compileDrop(t *testing.T) {
 		// Put existing contents except the top on stack
 		for i := 0; i < total-1; i++ {
 			loc := compiler.locationStack.pushValueLocationOnStack()
-			vm.valueStack[loc.stackPointer] = uint64(i) // Put the initial value.
+			ce.valueStack[loc.stackPointer] = uint64(i) // Put the initial value.
 		}
 
 		// Place the top value.
@@ -1805,9 +1805,9 @@ func TestArm64Compiler_compileCall(t *testing.T) {
 				require.Equal(t, builtinFunctionIndexGrowCallFrameStack, env.functionCallAddress(), env.functionCallAddress())
 
 				// Grow the callFrame stack, and exec again from the return address.
-				vm := env.callEngine()
-				vm.builtinFunctionGrowCallFrameStack()
-				jitcall(env.callFrameStackPeek().returnAddress, uintptr(unsafe.Pointer(vm)))
+				ce := env.callEngine()
+				ce.builtinFunctionGrowCallFrameStack()
+				jitcall(env.callFrameStackPeek().returnAddress, uintptr(unsafe.Pointer(ce)))
 			}
 
 			// Check status and returned values.
@@ -2014,9 +2014,9 @@ func TestArm64Compiler_compileCallIndirect(t *testing.T) {
 							require.Equal(t, builtinFunctionIndexGrowCallFrameStack, env.functionCallAddress(), env.functionCallAddress())
 
 							// Grow the callFrame stack, and exec again from the return address.
-							vm := env.callEngine()
-							vm.builtinFunctionGrowCallFrameStack()
-							jitcall(env.callFrameStackPeek().returnAddress, uintptr(unsafe.Pointer(vm)))
+							ce := env.callEngine()
+							ce.builtinFunctionGrowCallFrameStack()
+							jitcall(env.callFrameStackPeek().returnAddress, uintptr(unsafe.Pointer(ce)))
 						}
 
 						require.Equal(t, jitCallStatusCodeReturned, env.jitStatus())
@@ -2217,21 +2217,21 @@ func TestArm64Compiler_compileModuleContextInitialization(t *testing.T) {
 			require.Equal(t, jitCallStatusCodeReturned, env.jitStatus())
 
 			// Check if the fields of callEngine.moduleContext are updated.
-			vm := env.callEngine()
+			ce := env.callEngine()
 
 			bufSliceHeader := (*reflect.SliceHeader)(unsafe.Pointer(&tc.moduleInstance.Globals))
-			require.Equal(t, bufSliceHeader.Data, vm.moduleContext.globalElement0Address)
+			require.Equal(t, bufSliceHeader.Data, ce.moduleContext.globalElement0Address)
 
 			if tc.moduleInstance.MemoryInstance != nil {
 				bufSliceHeader := (*reflect.SliceHeader)(unsafe.Pointer(&tc.moduleInstance.MemoryInstance.Buffer))
-				require.Equal(t, uint64(bufSliceHeader.Len), vm.moduleContext.memorySliceLen)
-				require.Equal(t, bufSliceHeader.Data, vm.moduleContext.memoryElement0Address)
+				require.Equal(t, uint64(bufSliceHeader.Len), ce.moduleContext.memorySliceLen)
+				require.Equal(t, bufSliceHeader.Data, ce.moduleContext.memoryElement0Address)
 			}
 
 			if tc.moduleInstance.TableInstance != nil {
 				tableHeader := (*reflect.SliceHeader)(unsafe.Pointer(&tc.moduleInstance.TableInstance.Table))
-				require.Equal(t, uint64(tableHeader.Len), vm.moduleContext.tableSliceLen)
-				require.Equal(t, tableHeader.Data, vm.moduleContext.tableElement0Address)
+				require.Equal(t, uint64(tableHeader.Len), ce.moduleContext.tableSliceLen)
+				require.Equal(t, tableHeader.Data, ce.moduleContext.tableElement0Address)
 			}
 		})
 	}
@@ -4109,7 +4109,7 @@ func TestArm64Compiler_compileFConvertFromI(t *testing.T) {
 }
 
 func TestAmd64Compiler_compileBrTable(t *testing.T) {
-	requireRunAndExpectedValueReturned := func(t *testing.T, c *arm64Compiler, expValue uint32) {
+	requireRunAndExpectedValueReturned := func(t *testing.T, env *jitEnv, c *arm64Compiler, expValue uint32) {
 		// Emit code for each label which returns the frame ID.
 		for returnValue := uint32(0); returnValue < 7; returnValue++ {
 			label := &wazeroir.Label{Kind: wazeroir.LabelKindHeader, FrameID: returnValue}
@@ -4121,7 +4121,6 @@ func TestAmd64Compiler_compileBrTable(t *testing.T) {
 		}
 
 		// Generate the code under test and run.
-		env := newJITEnvironment()
 		code, _, _, err := c.compile()
 		require.NoError(t, err)
 		env.exec(code)
@@ -4266,7 +4265,7 @@ func TestAmd64Compiler_compileBrTable(t *testing.T) {
 
 			require.Len(t, compiler.locationStack.usedRegisters, 0)
 
-			requireRunAndExpectedValueReturned(t, compiler, tc.expectedValue)
+			requireRunAndExpectedValueReturned(t, env, compiler, tc.expectedValue)
 		})
 	}
 }

--- a/internal/wasm/jit/jit_other.go
+++ b/internal/wasm/jit/jit_other.go
@@ -11,7 +11,7 @@ type archContext struct{}
 
 func newArchContext() (ret archContext) { return }
 
-func jitcall(codeSegment, engine uintptr) {
+func jitcall(codeSegment, ce uintptr) {
 	panic("unsupported GOARCH")
 }
 

--- a/internal/wasm/jit/jit_test.go
+++ b/internal/wasm/jit/jit_test.go
@@ -12,7 +12,7 @@ import (
 
 type jitEnv struct {
 	eng            *engine
-	vm             *callEngine
+	ce             *callEngine
 	moduleInstance *wasm.ModuleInstance
 }
 
@@ -44,27 +44,27 @@ func (j *jitEnv) memory() []byte {
 }
 
 func (j *jitEnv) stack() []uint64 {
-	return j.vm.valueStack
+	return j.ce.valueStack
 }
 
 func (j *jitEnv) jitStatus() jitCallStatusCode {
-	return j.vm.exitContext.statusCode
+	return j.ce.exitContext.statusCode
 }
 
 func (j *jitEnv) functionCallAddress() wasm.FunctionIndex {
-	return j.vm.exitContext.functionCallAddress
+	return j.ce.exitContext.functionCallAddress
 }
 
 func (j *jitEnv) stackPointer() uint64 {
-	return j.vm.valueStackContext.stackPointer
+	return j.ce.valueStackContext.stackPointer
 }
 
 func (j *jitEnv) stackBasePointer() uint64 {
-	return j.vm.valueStackContext.stackBasePointer
+	return j.ce.valueStackContext.stackBasePointer
 }
 
 func (j *jitEnv) setStackPointer(sp uint64) {
-	j.vm.valueStackContext.stackPointer = sp
+	j.ce.valueStackContext.stackPointer = sp
 }
 
 func (j *jitEnv) addGlobals(g ...*wasm.GlobalInstance) {
@@ -80,19 +80,19 @@ func (j *jitEnv) setTable(table []wasm.TableElement) {
 }
 
 func (j *jitEnv) callFrameStackPeek() *callFrame {
-	return &j.vm.callFrameStack[j.vm.globalContext.callFrameStackPointer-1]
+	return &j.ce.callFrameStack[j.ce.globalContext.callFrameStackPointer-1]
 }
 
 func (j *jitEnv) callFrameStackPointer() uint64 {
-	return j.vm.globalContext.callFrameStackPointer
+	return j.ce.globalContext.callFrameStackPointer
 }
 
 func (j *jitEnv) setValueStackBasePointer(sp uint64) {
-	j.vm.valueStackContext.stackBasePointer = sp
+	j.ce.valueStackContext.stackBasePointer = sp
 }
 
 func (j *jitEnv) setCallFrameStackPointerLen(l uint64) {
-	j.vm.callFrameStackLen = l
+	j.ce.callFrameStackLen = l
 }
 
 func (j *jitEnv) module() *wasm.ModuleInstance {
@@ -104,7 +104,7 @@ func (j *jitEnv) engine() *engine {
 }
 
 func (j *jitEnv) callEngine() *callEngine {
-	return j.vm
+	return j.ce
 }
 
 func (j *jitEnv) exec(code []byte) {
@@ -117,11 +117,11 @@ func (j *jitEnv) exec(code []byte) {
 		},
 	}
 
-	j.vm.pushCallFrame(compiledFunction)
+	j.ce.pushCallFrame(compiledFunction)
 
 	jitcall(
 		uintptr(unsafe.Pointer(&code[0])),
-		uintptr(unsafe.Pointer(j.vm)),
+		uintptr(unsafe.Pointer(j.ce)),
 	)
 }
 
@@ -136,7 +136,7 @@ func newJITEnvironment() *jitEnv {
 			TableInstance:  &wasm.TableInstance{},
 			Globals:        []*wasm.GlobalInstance{},
 		},
-		vm: eng.newCallEngine(),
+		ce: eng.newCallEngine(),
 	}
 }
 

